x-pack/plugins/security_solution/public/timelines/containers/helpers.ts

```diff
@@ -5,9 +5,24 @@
  * 2.0.
  */
 
+import type { TimelineResult } from '../../../common/types';
 import { TableId } from '../../../common/types';
 import { DEFAULT_ALERTS_INDEX } from '../../../common/constants';
 
+export const getTimelineQueryTypes = (timeline: TimelineResult) => ({
+  hasQuery:
+    (timeline.kqlQuery != null &&
+      timeline.kqlQuery.filterQuery != null &&
+      timeline.kqlQuery.filterQuery.kuery != null &&
+      timeline.kqlQuery.filterQuery.kuery.expression !== '') ||
+    (timeline.dataProviders != null && timeline.dataProviders.length > 0) ||
+    (timeline.filters != null && timeline.filters.length > 0),
+  hasEql:
```

Comment:
Ah yea. FYI to anyone reading this, there's an existing bug where you can't delete an eql query so a timeline that has eql currently will always have eql. This should be fixed when this bug is resolved and has nothing to do with this work: https://github.com/elastic/kibana/issues/148950

Reply:
And slightly different issue on the KQL side if you don't actually submit your query as mentioned in https://github.com/elastic/kibana/pull/151091#issuecomment-1433772057.

Reply:
Yup! Autosave is a nice feature, but I feel like it's caused changes to be committed at times that people didn't want and other times led people to expect something to be saved that wasn't. We'll probably just add a save button :)